client/src/components/task-form/TaskForm.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { TASK_FORM_MODES } from 'constants/popup-modes';
+import { selectTaskToEdit, useTasksActions } from 'features/tasks/tasksSlice';
+import { useGlobalActions } from 'features/global/globalSlice';
+import { formatDateToISO } from 'utils/fortmatDateToISO';
+
+const TaskForm = () => {
+  const taskFormMode = useSelector((state) => state.global.taskFormMode);
+  const taskToEdit = useSelector(selectTaskToEdit);
+  const { addNewTask, updateTask } = useTasksActions();
+  const { setIsPopupOpen } = useGlobalActions();
+
+  const defaultValues = taskToEdit
+    ? {
+        title: taskToEdit.title,
+        description: taskToEdit.description,
+        deadline: taskToEdit.deadline
+          ? formatDateToISO(taskToEdit.deadline)
```

Comment:
Looks strange, but I think it's normal for this part

Reply:
Fine 👀